src/modules/submissions.js

```diff
@@ -2017,6 +2032,31 @@
 		}
 	}
 
+	function checkIfUserIsBlocked( userName ) {
+		var deferred = $.Deferred();
```

Comment:
using $.Deferred() is not needed here. You can just put a `return` in front of AFCH.api.get and and in front of `blockData` and `null` in the then/catch branches.